### PR TITLE
Media page

### PR DESCRIPTION
**Pull request by kumuthac23**

### CODE REVIEW

File: src/interface/types.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
+export interface IMedia {
+  id: string;
+  imageUrl: string;
+  title: string;
+  description: string;
+}
+export interface IMediaContent {
+  IMedia: string[];
```
Comment (praveensumathi) [inline src/interface/types.ts:8]: fix this

File: src/pageDrawers/CommonMediaDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  Grid,
+  TextField,
+  Typography,
+} from "@mui/material";
+import AddIcon from "@mui/icons-material/Add";
+import imageCompression from "browser-image-compression";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import CloseIcon from "@mui/icons-material/Close";
+
+function CommonMediaDrawer({ value, onClose, itemData, setItemData }) {
```
Comment (praveensumathi) [inline src/pageDrawers/CommonMediaDrawer.tsx:15]: create IProps

File: src/pageDrawers/CommonMediaDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    } else {
+      console.log(
+        "Invalid file format. Please select a JPEG or PNG or JPG file."
+      );
+    }
+  };
+
+  async function handleCompressFile(event: File): Promise<File | undefined> {
+    const imageFile = event;
+    console.log("originalFile instanceof Blob", imageFile instanceof Blob); // true
```
Comment (praveensumathi) [inline src/pageDrawers/CommonMediaDrawer.tsx:60]: remove the console

File: src/pageDrawers/CommonMediaDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    console.log(`originalFile size ${imageFile.size / 1024 / 1024} MB`);
```
Comment (praveensumathi) [inline src/pageDrawers/CommonMediaDrawer.tsx:61]: remove this

File: src/pages/home/Media.tsx [hunk 5]
```diff
@@ -1,7 +1,108 @@
...
+import {
+  MediaAchievementsData,
+  MediaEventData,
+} from "../../seedData/seed-data";
+import CommonMediaDrawer from "../../pageDrawers/CommonMediaDrawer";
+import { IMedia } from "../../interface/types";
+import CommonMediaTable from "../../common/components/CommonMediaTable";
 
 export default function Media() {
+  const [value, setValue] = useState("1");
```
Comment (praveensumathi) [inline src/pages/home/Media.tsx:17]: use enum

File: src/pages/home/Media.tsx [hunk 6]
```diff
@@ -1,7 +1,108 @@
...
+  const [isMediaDrawerOpen, setIsMediaDrawerOpen] = useState(false);
+  const [selectedData, setSelectedData] = useState<IMedia[]>([]);
+  const [selectedMediaId, setSelectedMediaId] = useState(null);
+  const [selectedMediaData, setSelectedMediaData] = useState<IMedia | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (value === "1") {
```
Comment (praveensumathi) [inline src/pages/home/Media.tsx:26]: use the enum values for the tab values

File: src/pages/home/Media.tsx [hunk 7]
```diff
@@ -1,7 +1,108 @@
...
+        </TabPanel>
+        <TabPanel value="2">
+          <CommonMediaTable
+            data={selectedData}
+            onEdit={handleMediaDrawerOpen}
+          />
+        </TabPanel>
+      </TabContext>
+
+      <Drawer
```
Comment (praveensumathi) [inline src/pages/home/Media.tsx:94]: move the drawers outside the page content

File: src/interface/types.ts (hunk 1 above)
Comment (kumuthac23) [reply src/interface/types.ts:8]: fixed

File: src/pageDrawers/CommonMediaDrawer.tsx (hunk 2 above)
Comment (kumuthac23) [reply src/pageDrawers/CommonMediaDrawer.tsx:15]: created IProps

File: src/pageDrawers/CommonMediaDrawer.tsx (hunk 4 above)
Comment (kumuthac23) [reply src/pageDrawers/CommonMediaDrawer.tsx:61]: console removed

File: src/pages/home/Media.tsx (hunk 5 above)
Comment (kumuthac23) [reply src/pages/home/Media.tsx:17]: enum created

File: src/pages/home/Media.tsx (hunk 6 above)
Comment (kumuthac23) [reply src/pages/home/Media.tsx:26]: enum value used

File: src/pages/home/Media.tsx (hunk 7 above)
Comment (kumuthac23) [reply src/pages/home/Media.tsx:94]: moved

File: src/pageDrawers/CommonMediaDrawer.tsx (hunk 3 above)
Comment (kumuthac23) [reply src/pageDrawers/CommonMediaDrawer.tsx:60]: console removed

File: src/pageDrawers/CommonMediaDrawer.tsx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+      <Box
+        sx={{
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+        }}
+      >
+        <Typography variant="h5" sx={{ p: 1 }}>
+          {itemData
+            ? `Edit ${value === "1" ? "Achievement" : "Event"}`
```
Comment (praveensumathi) [inline src/pageDrawers/CommonMediaDrawer.tsx:117]: use Enum values here

File: src/pageDrawers/CommonMediaDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+} from "@mui/material";
+import AddIcon from "@mui/icons-material/Add";
+import imageCompression from "browser-image-compression";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import CloseIcon from "@mui/icons-material/Close";
+
+interface IProps {
+  value: string;
+  onClose: () => void;
+  itemData: any;
```
Comment (praveensumathi) [inline src/pageDrawers/CommonMediaDrawer.tsx:18]: why this is any

**Review by praveensumathi — APPROVED** [review]

File: src/pageDrawers/CommonMediaDrawer.tsx (hunk 8 above)
Comment (kumuthac23) [reply src/pageDrawers/CommonMediaDrawer.tsx:117]: enum used

File: src/pageDrawers/CommonMediaDrawer.tsx (hunk 9 above)
Comment (kumuthac23) [reply src/pageDrawers/CommonMediaDrawer.tsx]: fixed

**Pull request merged**